implementations/rust/message/src/message.rs

```diff
@@ -645,120 +720,132 @@ mod tests {
-            0x8080,
-        ));
-        return_addresses.push(Address::UdpAddress(
-            AddressType::Udp,
-            IpAddr::V4(Ipv4Addr::new(10, 0, 1, 11)),
-            0x7070,
-        ));
-        return_addresses.push(Address::LocalAddress(
-            AddressType::Local,
-            LocalAddress {
-                address: 0x00010203,
-            },
-        ));
-        let onward_route = Route {
-            addresses: onward_addresses,
+        let mut onward_route = Route { addresses: vec![] };
+        let mut udp_address = Address::UdpAddress(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 0x8080);
+        let mut router_address = RouterAddress {
+            a_type: AddressType::Udp,
+            length: 0,
+            address: udp_address,
+        };
+        router_address.length = router_address.size_of();
+        onward_route.addresses.push(router_address);
+
+        let mut udp_address = Address::UdpAddress(IpAddr::V4(Ipv4Addr::new(10, 0, 1, 10)), 0x8090);
+        let mut router_address = RouterAddress {
+            a_type: AddressType::Udp,
+            length: 0,
+            address: udp_address,
         };
-        let return_route = Route {
-            addresses: return_addresses,
+        router_address.length = router_address.size_of();
+        onward_route.addresses.push(router_address);
+
+        let mut channel_address = Address::ChannelAddress(0x00010203);
+        let mut router_address = RouterAddress {
+            a_type: AddressType::Channel,
+            length: 0,
+            address: channel_address,
         };
+        router_address.length = router_address.size_of();
+        onward_route.addresses.push(router_address);
+
+        let mut return_route = Route { addresses: vec![] };
+        let mut udp_address = Address::UdpAddress(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 0x8080);
+        let mut router_address = RouterAddress {
+            a_type: AddressType::Udp,
+            length: 0,
+            address: udp_address,
+        };
+        router_address.length = router_address.size_of();
+        return_route.addresses.push(router_address);
+
+        let mut udp_address = Address::UdpAddress(IpAddr::V4(Ipv4Addr::new(10, 0, 1, 10)), 0x8090);
+        let mut router_address = RouterAddress {
+            a_type: AddressType::Udp,
+            length: 0,
+            address: udp_address,
+        };
+        router_address.length = router_address.size_of();
+        return_route.addresses.push(router_address);
+
+        let mut channel_address = Address::ChannelAddress(0x00010203);
+        let mut router_address = RouterAddress {
+            a_type: AddressType::Channel,
+            length: 0,
+            address: channel_address,
+        };
+        router_address.length = router_address.size_of();
+        return_route.addresses.push(router_address);
+
         let mut message_body = vec![0];
-        let mut msg = Message {
+        let mut msg = Box::new(Message {
```

Comment:
Wish I understood why Box is necessary.

Reply:
Don't you though. I addressed that earlier, and Box is gone inn the next PR.